lib/screenplay/config/permanent_config.ex

```diff
@@ -311,8 +314,14 @@ defmodule Screenplay.Config.PermanentConfig do
 
   defp update_existing_pending_screens(place_id, platform_ids, updated_pending_screens, acc) do
     Enum.reduce(updated_pending_screens, acc, fn
-      %{"is_deleted" => true, "screen_id" => screen_id}, acc ->
-        Map.delete(acc, screen_id)
+      %{"is_deleted" => true}, acc ->
+        # Only delete screen if "is_deleted" is true AND platform is at current place.
+        # If a screen ID is deleted at one place and moved to another,
+        # we need to make sure we are removing the correct screen.
+        Map.reject(acc, fn
+          %{alerts: %Alerts{stop_id: platform_id}} -> platform_id in platform_ids
```

Comment:
This doesn't look right, the function given to `Map.reject` should take a 2-tuple of key and value. Also, shouldn't we need to use `screen_id` somehow?

Reply:
Good catch. I was actually embarrassingly off. Fixed it so it now looks for platform ID and screen ID.